### PR TITLE
SwiftGen 6.5.0

### PR DESCRIPTION
**Pull request by djbe**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Release of https://github.com/SwiftGen/SwiftGen/pull/881

`brew audit --strict --online swiftgen` doesn't want to work, it keeps complaining about:
>  * HEAD: Use `branch: "stable"` to specify the default branch

Once I set it to `stable`, the audit passes. The problem is, our "head" branch is `develop`, not `stable`. In our [last release](https://github.com/Homebrew/homebrew-core/pull/62222) we set it to `develop`, but someone else [changed the formula](https://github.com/Homebrew/homebrew-core/commit/041a18c91312844f6aed3c79b1526f8007617c10) to point at `stable` for some reason.

### CODE REVIEW

**Comment by djbe** [comment]
Seems like there are 2 issues:
>   * Non-executables were installed to "/usr/local/opt/swiftgen/bin".
>    The offending files are:
>      /usr/local/opt/swiftgen/bin/SwiftGen_SwiftGenCLI.bundle
>  * Unexpected universal binaries were found.
>    The offending files are:
>      /usr/local/Cellar/swiftgen/6.5.0/bin/swiftgen

The first one is how SPM binaries with resources work, I don't think we're allowed to move those bundles to another place, it'll break the path of `Bundle.module`.

The second one, it's weird that only `11` is complaining and not `11-arm64` or any of the other tests, because we now always build universal binaries.

**Comment by branchvincent** [comment]
> it's weird that only 11 is complaining and not 11-arm64

ARM was being skipped, I've added the `CI-force-arm` label to correct that

> The second one is kinda minor

We can try the following:

```diff
diff --git a/Formula/swiftgen.rb b/Formula/swiftgen.rb
index 7f200071516..d8ec22ade49 100644
--- a/Formula/swiftgen.rb
+++ b/Formula/swiftgen.rb
@@ -53,6 +53,9 @@ class Swiftgen < Formula
       "#{fixtures}/Generated/XCAssets/swift5/all.swift"                   => "xcassets.swift",
       "#{fixtures}/Generated/YAML/inline-swift5/all.swift"                => "yaml.swift",
     })
+
+    # Replace universal binaries with their native slices
+    deuniversalize_machos
   end
 
   test do
```

> The first one is a much bigger issue. [...] Is there a way to silence this error in Homebrew?

Sorry, I'm honestly not sure. I'll let others chime in here.

**Comment by carlocab** [comment]
Passing a flag to the build to build only native binaries would be preferable to using `deuniversalize_machos`. (Though this should work too.)

Re the location of `SwiftGen_SwiftGenCLI.bundle`: let's install everything into `libexec` and then just symlink what we want to expose to users into `bin`, `lib`, etc.

**Comment by djbe** [comment]
I've updated the formula to install in `libexec` as suggested.

For now I'm also using `deuniversalize_machos`. It isn't ideal, but I'm working on an update to our build scripts, should be implemented in the next SwiftGen version.

File: Formula/swiftgen.rb [hunk 1]
```diff
@@ -2,105 +2,86 @@ class Swiftgen < Formula
...
-    }
-    (pkgshare/"fixtures").install fixtures
+    # Disable linting
+    ENV["NO_CODE_LINT"] = "1"
+
+    # Install SwiftGen in `libexec` (because of our resource bundle)
+    # Then create a symbolic link in `bin`
+    system "bundle", "exec", "rake", "cli:install[#{libexec}]"
+    Dir.mkdir bin
+    File.symlink "#{libexec}/swiftgen", "#{bin}/swiftgen"
```
Comment (carlocab) [inline Formula/swiftgen.rb:33]:
```suggestion
    bin.install_symlink libexec/"swiftgen"
```

Comment (djbe) [reply Formula/swiftgen.rb:33]: Did not know that existed, useful 😄

**Comment by djbe** [comment]
Nice the checks pass now 🎉 

Any thoughts on the `HEAD: Use branch: "stable" to specify the default branch` issue mentioned in the first comment?

**Comment by SMillerDev** [comment]
The default branch is `stable`, if that is not the branch that people should regard as the default branch when checking out the project I would recommend changing the default branch to `develop`.

**Comment by djbe** [comment]
The homebrew documentation mentions that `head` is for "the development cutting edge". Our `stable` branch only contains our releases, so `head` should point to `develop`.

My issue is that `brew audit --strict --online swiftgen` keeps complaining about this:
>  * HEAD: Use `branch: "stable"` to specify the default branch

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
> Nice the checks pass now 🎉
> 
> Any thoughts on the `HEAD: Use branch: "stable" to specify the default branch` issue mentioned in the first comment?

@nandahkrishna, thoughts here? (I think you implemented this audit -- apologies if I'm misremembering.)

**Comment by djbe** [comment]
From the git history for the formula, seems like @chenrui333 made the change to `stable` if that helps?

**Pull request closed without merging**